### ISSUE TITLE
License of cpp extension stating MIT but the content is GPL

### ISSUE THREAD

**Issue by Mingpan** [microsoft/vscode#193966]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.82.2
- OS Version: irrelevant

Steps to Reproduce: https://github.com/microsoft/vscode/blob/1.82.2/ThirdPartyNotices.txt#L776

The title states MIT License, but the content text appears to be GPL.

**Comment by Mingpan** [comment]
Relevant context: https://github.com/microsoft/vscode/commit/eb03486765b6e06f82e392fd7f7da3f5a0c0e144

**Comment by isidorn** [comment]
Hey @Mingpan 👋  thanks for opening this issue.
We previously had discussion about this and I believe the license is MIT.

@alexr00 can you confirm? Also the bug is probably that our License generation tool fetches the license from the linked Repo https://github.com/jeff-hykin/better-cpp-syntax and looking at it there it is indeed GPL.

**Comment by alexr00** [comment]
The license at the commit that we reference in that repo is MIT:

https://github.com/jeff-hykin/better-cpp-syntax/blob/f1d127a8af2b184db570345f0bb179503c47fdf6/LICENSE

The bug in indeed that the generation tool fetches the license from the linked repo, but not at the relevant commit.

**Comment by isidorn** [comment]
Makes sense. I think we can do the following independent things:

1. Manually fix this instance by explicitly putting in the `licenseDetail` in the `cgmanifest.json` file. Like this component does https://github.com/microsoft/vscode/blob/4b1f945bdf49ffe67d6bb603128455661140aaf5/extensions/cpp/cgmanifest.json#L38
2. Look into the tool to respect the commit hash

What do you think?

**Comment by Mingpan** [comment]
Thanks Isidor and Alex for the quick response!

One follow-up question: for these cases, would license statement inside `cgmanifest.json` file also be synced into the [ThirdPartyNotices.txt](https://github.com/microsoft/vscode/blob/1.82.2/ThirdPartyNotices.txt)?

For example, src/vs/workbench/services/textMate/common/cgmanifest.json also specifies a `licenseDetail`, but I can't find the corresponding paragraph inside `ThirdPartyNotices.txt`. Or am I missing something?

**Comment by isidorn** [comment]
@Mingpan yes, the licenseDetail should be transferred to a corresponding part in `thirdpartynotices.txt`.
@alexdima do you still own the tool to generate notices?
